### ISSUE TITLE
Support OSS LLMs

### ISSUE THREAD

**Issue by homanp** [homanp/superagent#35]

**Comment by bdqfork** [comment]
Perhaps we can export the environment variable OPENAI_ BASE_ URL to support projects such as llama-cpp-python to suppoer OSS LLMs ?

**Comment by homanp** [comment]
> Perhaps we can export the environment variable OPENAI_ BASE_ URL to support projects such as llama-cpp-python to support OSS LLMs ?

Sure, contribs would be nice here!

**Comment by homanp** [comment]
> Here, OPENAI_ BASE_ URL works well. Just set this env, no need to modify code.

What if one wants to run gpt4all etc?

**Comment by bdqfork** [comment]
The majority of OSS models can now be deployed through HuggingFace, which we already support. GPT4All simply provides an alternative ecosystem for loading OSS models, the model is still loaded based on other methods. Perhaps we should consider how to support Llama.cpp and ExLlama as providers directly?

**Comment by homanp** [comment]
> > > Here, OPENAI_ BASE_ URL works well. Just set this env, no need to modify code.
> 
> > 
> 
> > What if one wants to run gpt4all etc?
> 
> 
> 
> 
> 
> The majority of OSS models can now be deployed through HuggingFace, which we already support. GPT4All simply provides an alternative ecosystem for loading OSS models, the model is still loaded based on other methods. Perhaps we should consider how to support Llama.cpp and ExLlama as providers directly?

You are right. I made PR on gpt4all a month back but found it not fit to run in prod.
